packages/react/src/domain/types.ts

```diff
@@ -110,3 +125,11 @@ export type ValidationError = {
 export type Validator<T = unknown> = (value: T) => ValidationError | undefined;
 
 export type MaybeArray<T> = T | T[];
+
+export function isFactorWithAllowedHandleTypes(
+  factor: Factor
+): factor is FactorWithAllowedHandleTypes {
+  return Object.hasOwn(factor, "allowedHandleTypes");
```

Comment:
Should we assert this is an array with the correct items? This value comes from the developer, so it may be incorrect.